OrangeDesignSystem/Sources/OrangeDesignSystem/Components/Lists/ODSListItem.swift

```diff
@@ -0,0 +1,289 @@
+///                 }
+///             }
+///         }
+///     }
+///
+/// - selection: composed of leading icons, title, subtile and trailing indicator or switch to select the element.
+/// Click on the list item with select or unselect the item.
+///
+/// The following example shows how to build a List with selection elements:
+///
+///     struct YourView: View {
+///         @State private var optionA: Bool
+///         @State private var optionB: Bool
+///
+///         var body: some View {
+///             NavigationView {
+///                 List {
+///                    ODSListItem(
+///                         title: Text("Option A"),
+///                         subtitle: Text("Option A description"),
+///                         trailingCheckmarkIsSelected: optionA
+///                    )
+///                    .odsListItemStyle()
+///                    .onTapGesture {
+///                         optionA.toggle()
+///                    }
+///
+///                    ODSListItem(
+///                         title: Text("Option B"),
+///                         subtitle: Text("Option B description"),
+///                         trailingCheckmarkIsSelected: optionB)
+///                    )
+///                    .odsListItemStyle()
+///                    .onTapGesture {
+///                         optionB.toggle()
+///                    }
+///                 }
+///             }
+///         }
+///     }
+///
+/// REMARK: Don't forget top apply the style __.odsListItemStyle__ to get right frame height and paddings.
+///
+public struct ODSListItem: View {
+
+    /// Describes the leading icon of the list item.
+    ///
+    /// Images can be loaded from resources or asynchronously from the specified URL.
+    /// Until the image loads, the placeholder image is displayed.
+    /// Image and placeholder are shaped (circular, wide or square)
+    ///
+    /// Icons are loaded from ressources and no shape is applyed.
+    ///
+    public enum Leading {
+        case icon(Image) // A Solaris icon from resource.
+        case circularImage(source: ODSImage.Source) // An image cropped into a circle.
+        case squareImage(source: ODSImage.Source) // An image cropped into a square.
+        case wideImage(source: ODSImage.Source) // An image cropped into a rectangle.
+    }
+
+    /// Describes the trailing element of the list item.
+    /// This type of element can be an information, a selection or an action
+    ///
+    enum TrailingElement {
+        case textOnly(Text) // A text
+        case iButton(() -> Void, Text?) // A i button with its action and an optinal text.
+        case toggle(Binding<Bool>) // A toggle with its binding that determines whether it is on or off.
+        case checkmark(Bool) // A checkmark with a flag to indicate if it is visbile or not.
+    }
+
+    // =======================
+    // MARK: Stored Properties
+    // =======================
+
+    private let title: Text
```

Comment:
Are you sure storing in properties `Text` is better than storing String objects?

Reply:
FI, using `Text` here is more suitable because SDK customers must provide their `Text` to the component we implemented. By receiving a `Text`we will be able to qdd constrqints on it to prevent customers to chqnge colors, fonts or sizes for examples. SDK users will have to deal with their own stuff about a11y, not our side.